python/ml4ir/applications/ranking/tests/test_lr_schedule.py

```diff
@@ -0,0 +1,223 @@
+import unittest
+import warnings
+import os
+from ml4ir.base.io import logging_utils
+from ml4ir.base.io.local_io import LocalIO
+from ml4ir.base.features.preprocessing import *
+from ml4ir.base.model.optimizers import cyclic_learning_rate
+from ml4ir.base.config.keys import DataFormatKey, TFRecordTypeKey
+from ml4ir.base.data.relevance_dataset import RelevanceDataset
+from ml4ir.base.model.relevance_model import RelevanceModel
+from ml4ir.base.model.losses.loss_base import RelevanceLossBase
+from ml4ir.base.model.scoring.scoring_model import ScorerBase, RelevanceScorer
+from ml4ir.base.model.scoring.interaction_model import InteractionModel, UnivariateInteractionModel
+from ml4ir.base.model.optimizers.optimizer import get_optimizer
+from ml4ir.applications.ranking.model.ranking_model import RankingModel
+from ml4ir.applications.ranking.config.keys import LossKey
+from ml4ir.applications.ranking.config.keys import ScoringTypeKey
+from ml4ir.applications.ranking.model.losses import loss_factory
+import yaml
+from ml4ir.base.features.feature_config import ExampleFeatureConfig, SequenceExampleFeatureConfig
+import tensorflow as tf
+from tensorflow import keras
+import numpy as np
+from tensorflow.keras.optimizers.schedules import ExponentialDecay
+
+warnings.filterwarnings("ignore")
+
+INPUT_DIR = "ml4ir/applications/ranking/tests/data/"
+MODEL_CONFIG = "ml4ir/applications/ranking/tests/data/configs/model_config_cyclic_lr.yaml"
+KEEP_ADDITIONAL_INFO = 0
+NON_ZERO_FEATURES_ONLY = 0
+
+class LrCallback(keras.callbacks.Callback):
+    """Defining a callback to keep track of changing learning rate during training."""
+
+    def __init__(self):
+        self.lr_list = []
+
+    def on_train_batch_begin(self, batch, logs=None):
+        self.lr_list.append(self.model.optimizer._decayed_lr(tf.float32).numpy())
+
+    def get_lr_list(self):
+        return self.lr_list
+
+class TestLrSchedules(unittest.TestCase):
+    """Testing different learning rate schedules"""
+
+    def setUp(self):
+        self.feature_config_yaml_convert_to_clicks = INPUT_DIR + \
+            'ranklib/feature_config_convert_to_clicks.yaml'
+        self.model_config_file = MODEL_CONFIG
+
+    def compare_lr_values(self, scheduler, expected_values):
+        """Expects a learning rate scheduler `scheduler` and
+        a list of values. Compares ...."""
```

Comment:
typo in `....`?

Reply:
Also, the docstring is not accurate, it currently does not get a scheduler but the lrs of the scheduler I think.